hack/codespell.sh

```diff
@@ -0,0 +1,20 @@
+#! /usr/bin/env bash
+
+readonly PROGNAME="codespell"
+
+if command -v ${PROGNAME} >/dev/null; then
+	exec ${PROGNAME} "$@"
+fi
+
+if command -v docker >/dev/null; then
+	exec docker run --rm -i \
+		-v $(pwd):/workdir \
+		lianghao208/codespell:v2.1.0 "$@"
```

Comment:
we should be using official/library images as much as we can
looking at https://hub.docker.com/r/lianghao208/codespell, looks like it was just created for this PR
without having access to the Dockerfile that created this image and this image being used by the larger community in general, im less inclined to allow this to be checked in.
We should either skip `codespell`, or go with [one](https://github.com/envoyproxy/gateway/issues/90#issue-1259340650) to install tooling
cc @danehans @youngnick

Reply:
Can we instruct developers to install `codespell` directly from `https://github.com/codespell-project/codespell#installation` if they haven't installed yet, instead of running a docker container for them? 

In that case, we can simply remove the `docker run` step from hack shell.

xref: [contour/codespell](https://github.com/projectcontour/contour/blob/main/hack/codespell.sh)

Reply:
@danehans above steps of asking users to manually install `codespell` is not a great dev experience. This is why [one](https://github.com/envoyproxy/gateway/issues/90#issue-1259340650) has been a very pattern adopted by many projects.

Reply:
yah based on the learnings from this PR, I prefer moving to 1. Im happy to start the work for it

Reply:
@arkodg I'm fine with pivoting to #1. Please update #90 and cc the other maintainers to ensure everyone's on the same page.